src/main/resources/public/ts/sniplets/structure-option-is-average-skills.ts

```diff
@@ -0,0 +1,46 @@
+
+import {structureOptionsService, IStructureOptionsService, StructureOptions} from "../services";
+
+import {toasts} from "entcore";
+import {safeApply} from "../utils/functions/safeApply";
+
+export class SnipletStructureOptions {
+   option : StructureOptions;
+
+    constructor( private $scope: any, private structureOptionsService: IStructureOptionsService) {
+        this.initOptionsIsAverageSkills($scope.source.idStructure);
+    }
+
+    async initOptionsIsAverageSkills(structureId: String): Promise<void> {
+        try {
+            let option = await structureOptionsService.getStructureOptionsIsAverageSkills(structureId);
```

Comment:
Tu peux le faire en une ligne non ?
```suggestion
      this.option = await structureOptionsService.getStructureOptionsIsAverageSkills(structureId);
```

Reply:
Attention au typage!